### DISCUSSION TITLE
Testing/Dev Framework to Improve Plugin Developer Experience

### DISCUSSION THREAD

**Discussion by nabeelvalley** [strapi/strapi#6274] [Legacy Topics]
The current plugin development system is constantly changing and I feel it would be beneficial for plugin authors to have some sort of framework that can be used to improve the plugin experience and a way for these plugins to be tested without the need for a manual Strapi end-to-end test to be done for plugin development that ensures plugins are aligned with Strapi updates/versions

Essentially a testing library/framework that plugin developers can use to build plugins

**Comment by lauriejim** [comment]
Thank you for this insight, I ping @Aurelsicoko @soupette and @alexandrebodin for this topic discussion.

Here is a guide for unit testing - https://github.com/strapi/strapi/pull/6324
